platforms/core-configuration/file-collections/src/main/java/org/gradle/api/internal/file/collections/DefaultConfigurableFileCollection.java

```diff
@@ -472,6 +481,11 @@ public void replace(Transformer<? extends @org.jetbrains.annotations.Nullable Fi
         }
     }
 
+    @Override
+    public void attachOwner(@Nullable ModelObject owner, DisplayName displayName) {
```

Comment:
💭 This change probably attach owner for `ConfigurableFileCollection` only in the case it's `abstract ConfigurableFileCollection gettter()`, since for other cases we don't attach owner for ConfigurableFileCollection, e.g.:
https://github.com/gradle/gradle/blob/6611f07460222cb47d254aa3015608ae7ea8da4a/platforms/core-configuration/model-core/src/main/java/org/gradle/internal/instantiation/generator/AbstractClassGenerator.java#L423-L425

and

https://github.com/gradle/gradle/blob/6611f07460222cb47d254aa3015608ae7ea8da4a/platforms/core-configuration/model-core/src/main/java/org/gradle/internal/instantiation/generator/AbstractClassGenerator.java#L435-L437

Reply:
Not sure what are `for backwards compatibility` concerns

Reply:
:x: Let's explain this in the code as well by adding a comment.

Reply:
There is already a comment:
```
// This should apply to all 'managed' types however for backwards compatibility is applied only to property types ```
Or you had something else in mind?

Do you maybe know what these `backwards compatibility` concerns are or we should just implement it?